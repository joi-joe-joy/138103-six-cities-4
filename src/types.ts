export enum HouseType {
  APARTMENT = `apartment`,
  ROOM = `room`,
  HOUSE = `house`,
  HOTEL = `hotel`
};

export enum PlaceCardType {
  CITIES = `cities`,
  NEAR = `near`,
  FAVORITES = `favorites`,
  PROPERTY = `property`
};

export enum PageType {
  PROPERTY = `property`,
  MAIN = `main`,
  LOGIN = `login`,
  FAVORITES = `favorites`,
  FAVORITES_EMPTY = `favorites_empty`
};

export enum SortType {
  POPULAR = `popular`,
  TO_HIGH = `to-high`,
  TO_LOW = `to-low`,
  TOP_RATED = `top-rated`
};

export enum AuthStatus {
  NO_AUTH = `NO_AUTH`,
  AUTH = `AUTH`,
};

export interface Location {
  latitude: number,
  longitude: number,
  zoom: number
}

export interface City {
  location: Location,
  name: string
}

export interface User {
  id: number,
  avatarUrl: string,
  name: string,
  isPro: boolean
}

export type AuthInfo = User & {
  email: string
}

export interface Offer {
  id: number,
  title: string,
  maxAdults: number,
  description: string,
  isPremium: boolean,
  isFavorite: boolean,
  images: Array<string>,
  previewImage: string,
  goods: Array<string>,
  price: number,
  rating: number,
  bedrooms: number,
  type: HouseType.APARTMENT | HouseType.ROOM | HouseType.HOUSE | HouseType.HOTEL,
  host: User,
  location: Location,
  city: City,
};

export interface Comment {
  comment: string,
  date: string,
  id: number,
  rating: number,
  user: User
}
