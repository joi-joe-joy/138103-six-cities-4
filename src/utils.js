
export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const noop = () => {
  // do nothing for test
};

// Checking nested objects
const processValue = (value) => {
  if (typeof value !== `object` || value === null) {
    return value;
  } else {
    return Array.isArray(value)
      ? value.map(processValue)
      : renameKeys(value);
  }
};

// Rename Object Keys from Snake Case to Camel Case
export const renameKeys = (object) => {
  return Object.fromEntries(
      Object.entries(object)
        .map(([key, value]) => [
          key.replace(/_(.)/g, (word) => word[1].toUpperCase()),
          processValue(value)
        ])
  );
};
