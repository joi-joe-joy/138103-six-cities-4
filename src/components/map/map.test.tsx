import * as React from "react";
import * as renderer from "react-test-renderer";
import {PageType} from "../../types";
import Map from "./map";
import {noop} from "../../utils";

jest.mock(`leaflet`);

describe(`Render Map`, () => {
  it(`Renderer Map correctly`, () => {
    const tree = renderer.create(
        <Map
          type={PageType.PROPERTY}
        >
          <div/>
        </Map>, {
          createNodeMock: noop
        }
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Renderer Map correctly with current`, () => {
    const tree = renderer.create(
        <Map
          type={PageType.MAIN}
        >
          <div/>
        </Map>, {
          createNodeMock: noop
        }
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
