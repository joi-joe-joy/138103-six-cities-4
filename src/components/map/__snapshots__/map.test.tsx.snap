// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Map Renderer Map correctly 1`] = `
<section
  className="map property__map"
  style={
    Object {
      "width": "100%",
    }
  }
>
  <div />
</section>
`;

exports[`Render Map Renderer Map correctly with current 1`] = `
<section
  className="map cities__map"
  style={
    Object {
      "width": "100%",
    }
  }
>
  <div />
</section>
`;
